Mapping/LiSim/sim.py

```diff
@@ -37,25 +37,26 @@ def save_carte(self):
     def give_carte(self):
         return self.carte
 
-    def noise(self, clean_data):
+    def _noise(self, clean_data, dev, esp=0):
         '''Function that adds noise from a normal distribution to the data provided to mimic real world'''
         data = np.array(clean_data)
-        noise = np.random.normal(self.mu,self.std,[int(360/self.angle_step),2])
+        noise = np.random.normal(esp,dev,clean_data.shape)
         return noise+clean_data
 
-    def simulate(self, show=False):
+    def simulate(self, show=False, uPos=2, uDist=2.5, uTheta=1.25):
         '''Simulates data from one complete lidar rotation'''
         data=[]
         if show == True:
             print(self.carte)
         for i in range(int(360/self.angle_step)):
             current_ray = 1
+            theta = i
```

Comment:
Why not use i?

Reply:
Because it’s an iterator over the data
Here you assume all is in good order and you have 1* per measurement
In a real case scenario that won’t be the case, so it’s “setting the ground”

Reply:
Ok nice